twill-api/src/main/java/org/apache/twill/internal/DefaultTwillRunResources.java

```diff
@@ -90,8 +104,21 @@ public Integer getDebugPort() {
   }
 
   @Override
-  public Level getLogLevel() {
-    return logLevel;
+  public Level getRootLogLevel() {
+    return rootLogLevel;
+  }
+
+  @Override
+  public Map<String, LogEntry.Level> getLogLevelArguments() {
+    return logLevelArguments;
+  }
+
+  @Override
+  public void updateLogLevel(Map<String, Level> logLevelArguments) {
```

Comment:
Better keep this class immutable. If there is a change in the log level, simply create a new instance.